File: Prixly/Prixly.Dal/Models/TransactionLogDto.cs
using Prixly.Dal.Enums;

namespace Prixly.Dal.Models
{
    public class TransactionLogDto
    {
        public Guid Guid { get; set; }
        public Guid UserGuid { get; set; }
        public DateTime Timestamp { get; set; }
        public decimal Amount { get; set; }
        public MerchantCategoryCode Category { get; set; }
        public CreditCardDto SelectedCard { get; set; }
        public decimal EarnedReward { get; set; }
    }
}

File: Prixly/Prixly.Client/src/models/transaction-log-dto.ts
import { MerchantCategoryCode } from './enums/merchant-category-code.enum';
import { CreditCardDto } from './credit-card.dto';

export interface TransactionLogDto {
  guid: string;
  userGuid: string;
  timestamp: string; // ISO format
  amount: number;
  category: MerchantCategoryCode;
  selectedCard: CreditCardDto;
  earnedReward: number;
}
